src/serac/physics/utilities/functional/functional.hpp

```diff
@@ -420,12 +395,13 @@ class Functional<test(trial)> : public mfem::Operator {
         }
       }
 
+      output_L_boundary_ = 0.0;
+
       // scatter-add to compute residuals on the local processor
       G_test_boundary_->MultTranspose(output_E_boundary_, output_L_boundary_);
```

Comment:
After poking around, I believe this does not work in parallel due to this operator application. It looks like MFEM is not treating shared boundaries in the DOF numbering appropriately, causing the boundary contributions to be added at the wrong index.

Reply:
As this could possibly be a rabbit-hole chasing down a bug in MFEM, I propose we mark this as a TODO for now and merge it in as-is so we can continue functional development in parallel.